helix-core/src/main/java/org/apache/helix/controller/rebalancer/util/DelayedRebalanceUtil.java

```diff
@@ -137,14 +138,13 @@ private static long getInactiveTime(String instance, Set<String> liveInstances,
     }
 
     // check the time instance got disabled.
-    if (!instanceConfig.getInstanceEnabled() || (clusterConfig.getDisabledInstances() != null
-        && clusterConfig.getDisabledInstances().containsKey(instance))) {
+    if (!InstanceValidationUtil.isInstanceEnabled(instanceConfig, clusterConfig)) {
       long disabledTime = instanceConfig.getInstanceEnabledTime();
-      if (clusterConfig.getDisabledInstances() != null && clusterConfig.getDisabledInstances()
-          .containsKey(instance)) {
+      Map<String, String> disabledInstances = clusterConfig.getDisabledInstances();
+      if (disabledInstances.containsKey(instance)) {
```

Comment:
We could use clusterConfig.getInstanceHelixDisabledTimeStamp() ?

Reply:
Could you please elaborate? What specifically are you proposing to change?

Reply:
Here we read from clusterConfig to get getDisabledInstances and parse target instance's disabled info to get timestamp. 
There is a getter in clusterConfig to get disabled time (getInstanceHelixDisabledTimeStamp). We could use that directly.

Reply:
Thanks, updated